DESIGN_DOCUMENT_TEMPLATE.md

```diff
@@ -0,0 +1,139 @@
+- [Using this Document](#using-this-document)
+	- [Feature Description](#feature-description-)
+	- [Architecture Diagram](#architecture-diagram)
+		- [API Diagram](#api-diagram)
+	- [Documenting your Feature](#documenting-your-feature)
+	- [Security Reviews](#security-reviews)
+
+# Using this Document
+
+This document provides an easy-to-use template to help you get started building your own design document for OpenSearch features.
+
+To use this document you should:
+
+1. Copy and paste the contents of the document to a new file <YOUR_FEATURE>.md or to the body of a GitHub comment.
+2. Fill out the remainder of this document.
+
+**Please do not overwrite this document.**
+
+## Feature Description
```

Comment:
I feel like there are a couple of purposes for a doc like this, are we trying to explain for broad understanding - so the feature and diagrams should be about user facing experience and user flows, or is this about technical implementations, this feature creates a token with properties x, y, & z, it will have Rest APIs ,POST /_security/foo/token { ... }, and it will add functionality in class files AFactory, BSingleton, and CInstance.

I think both cases have their use, but the audience would be different.

Reply:
If you are trying to provide guidance for creating a 'meta' issue, such as recreation https://github.com/opensearch-project/security/issues/2573 I think focusing in on user scenarios and then project management scenarios might make more sense.  I would also love a template for our meta issues :D

Reply:
On the other hand, if it is more of an RFC on a prototype/spike, I think that might take a more technical format - and be associated with a higher level meta issues.

Reply:
I am gonna go with a more technical one for now since it seems more in line with the rest of the document.